docs: Translation Flow (#40737)
freeCodeCamp/freeCodeCamp: docs/how-to-translate-files.md

To begin, head to our translation website and login (if you have not contributed to translations before, you will need to create an account).

## Select a Project and File

You should see two "projects" available for translation: The `Contributing Documentation` project, which contains the files for this documentation site, and the `Coding Curriculum` project, which contains our challenge files for `/learn`.

Select which project you want to contribute to, and you will see a list of available languages for translation.

![Image - List of available languages](./images/crowdin/languages.png)

Select the language you want to work on, and you will see the complete file tree.

![Image - List of available files](./images/crowdin/file-tree.png)

Each file and folder will show a progress bar. The **blue** portion of the progress bar indicates what percentage of the file has been translated, while the **green** portion of the progress bar indicates what percentage of the file has been approved by the proofreading team.

Select a file to work on and Crowdin will open the editor view.

## Translate the File

![Image - Editor View](./images/crowdin/editor.png)

> Crowdin separates a document into translatable "strings", usually as sentences. Each string is translated individually. Referring to the image above:

Reviewer: ```suggestion
Crowdin separates a document into translatable "strings", usually sentences. Each string is translated individually. Referring to the image above:
```

1. A string highlighted in green already has a proposed translation.
2. A string highlighted in red does *not* have a proposed translation.

> 3. A string with greyed out text is not translatable. This is the case for code blocks and other content that must be preserved in English. You will be unable to select these strings in the editor.

Reviewer: ```suggestion
3. A string with greyed out text is not translatable. This is the case for code blocks and other content that must not be translated. You will be unable to select these strings in the editor.
```
The original text sounded to me as if the code blocks were in English.

5. Crowdin will recommend translations based on Translation Memory (TM) or Machine Translation (MT). Translation Memory refers to similar or identical strings that we have translated/approved in other files. Machine Translation refers to translations recommended by their integrated library.
6. This is the editor pane, where you may write your proposed translation for the selected string.
7. The currently selected string in the editor will be highlighted in yellow.
8. Here you will see tags indicating the state of the string. `Done` means the string has at least one proposed translation. `Todo` means the string does not have any proposed translations.
9. Here you can see the comments window. If you have questions or concerns about a particular string, you can leave a comment on the string here for other translators to see.
10. These two "pane" buttons will hide the left (document) and right (comments) views.

> [!NOTE]

> > If you see a hidden string that includes translations, please notify us in the [contributors chat room](https://chat.freecodecamp.org/contributors) so we can remove the translation from memory.

Reviewer: Just checking, is this the workflow?  I would have thought we want to reduce friction and get reports through Crowdin.

Reviewer 2: Because the strings are hidden, translators can't select them in the Crowdin editor and are unable to leave a comment directly on the string.

And I did not see a way to leave comments on the general file - we could have folks add comments on the string before/after the hidden one, but that might get confusing.

Reviewer: Thanks, I understand the problem now.  I did find a way to comment on the file, though: 

![image](https://user-images.githubusercontent.com/15801806/105045225-f694a480-5a67-11eb-9d43-f2be4ae26014.png)

This is on my non-admin account.

Reviewer 2: That is the comment flow, yes - but my understanding is that comment will be applied to whatever string you have selected (I believe, when you initially open the file, it's the first selectable string).

Reviewer: Oops, I opened a file and assumed that was a generic comment section.  You're right - it's just the first string.

Thanks for bearing with me.

> When you have completed a translation for a string, select the `Save` button to store your translation on Crowdin. Other contributors will then be able to vote your translation as accurate and proofreaders will be able to approve it.

Reviewer: ```suggestion
When you have completed a translation for a string, select the `Save` button to store your translation on Crowdin. Other contributors will then be able to vote on your translation and proofreaders will be able to approve it.
```
